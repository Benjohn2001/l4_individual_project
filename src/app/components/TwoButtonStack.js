import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text, View } from "react-native";

function TwoButtonStack(props) {
  const { title1, onPress1, title2, onPress2 } = props;
  return (
    <View className="flex-col space-y-2">
      <TouchableOpacity
        onPress={onPress1}
        className="bg-primaryPurple h-12 w-52 pt-3 rounded-full mx-1.5"
      >
        <Text className="text-darkerPurple text-sm text-center">{title1}</Text>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={onPress2}
        className="bg-darkerPurple h-12 w-52 pt-3 rounded-full mx-1.5"
      >
        <Text className="text-white text-sm text-center">{title2}</Text>
      </TouchableOpacity>
    </View>
  );
}

TwoButtonStack.propTypes = {
  title1: PropTypes.string.isRequired,
  onPress1: PropTypes.func.isRequired,
  title2: PropTypes.string.isRequired,
  onPress2: PropTypes.func.isRequired,
};

export default TwoButtonStack;
