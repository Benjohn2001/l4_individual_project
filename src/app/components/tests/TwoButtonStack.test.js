import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";
import TwoButtonStack from '../TwoButtonStack';

test('Check TwoButtonStack renders correctly and onpress works',()=>{
    const onPressMock1 = jest.fn();
    const onPressMock2 = jest.fn();
    render(
        <TwoButtonStack 
            title1="press1"
            onPress1={()=>{
                onPressMock1()
            }}
            title2="press2"
            onPress2={()=>{
                onPressMock2()
            }}
        />,
      );
      fireEvent.press(screen.getByText('press1'));
      expect(onPressMock1)
      fireEvent.press(screen.getByText('press2'));
      expect(onPressMock2)
      const domTree = renderer.create(
        <TwoButtonStack 
            title1="press1"
            onPress1={()=>{
                onPressMock1()
            }}
            title2="press2"
            onPress2={()=>{
                onPressMock2()
            }}
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})