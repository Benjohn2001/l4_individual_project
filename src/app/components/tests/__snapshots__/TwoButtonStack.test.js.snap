// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check TwoButtonStack renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#C6C4FF",
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "height": 48,
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
        "paddingTop": 12,
        "width": 208,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B4EFF",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 20,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      press1
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6B4EFF",
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
        "height": 48,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 8,
        "opacity": 1,
        "paddingTop": 12,
        "width": 208,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 14,
            "lineHeight": 20,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      press2
    </Text>
  </View>
</View>
`;
